File: django_airavata/apps/workspace/urls.py

from django.urls import re_path

from . import views

app_name = 'django_airavata_workspace'
urlpatterns = [
    re_path(r'^projects$', views.projects_list, name='projects'),
    re_path(r'^projects/(?P<project_id>[^/]+)/$', views.edit_project,
            name='edit_project'),
    re_path(r'^experiments/(?P<experiment_id>[^/]+)/edit$', views.edit_experiment,
            name='edit_experiment'),
    re_path(r'^experiments/(?P<experiment_id>[^/]+)/$', views.view_experiment,
            name='view_experiment'),
    re_path(r'^experiments$', views.experiments_list, name='experiments'),
    re_path(r'^applications/(?P<app_module_id>[^/]+)/create_experiment$',
            views.create_experiment, name='create_experiment'),
    re_path(r'^dashboard$', views.dashboard, name='dashboard'),
    re_path(r'^storage', views.user_storage, name='storage'),
]

File: django_airavata/apps/workspace/static/django_airavata_workspace/js/utils/urls.js
export default {
  editExperiment(experiment) {
    return (
      "/workspace/experiments/" +
      encodeURIComponent(experiment.experimentId) +
      "/edit"
    );
  },
  navigateToEditExperiment(experiment) {
    window.location.assign(this.editExperiment(experiment));
  },
  experimentsList() {
    return "/workspace/experiments";
  },
  navigateToExperimentsList() {
    window.location.assign(this.experimentsList());
  },
  viewExperiment(experiment, { launching = false } = {}) {
    return (
      "/workspace/experiments/" +
      encodeURIComponent(experiment.experimentId) +
      "/" +
      (launching ? "?launching=true" : "")
    );
  },
  navigateToViewExperiment(experiment, { launching = false } = {}) {
    window.location.assign(
      this.viewExperiment(experiment, { launching: launching })
    );
  },
  createExperiment(appModule) {
    return (
      "/workspace/applications/" +
      encodeURIComponent(appModule.appModuleId) +
      "/create_experiment"
    );
  },
  navigateToCreateExperiment(appModule) {
    window.location.assign(this.createExperiment(appModule));
  },
  editProject(project) {
    return "/workspace/projects/" + encodeURIComponent(project.projectID) + "/";
  },
  projectsList() {
    return "/workspace/projects";
  },
  navigateToProjectsList() {
    window.location.assign(this.projectsList());
  },
};
